vortex-array/src/compute/sum.rs

```diff
@@ -43,14 +43,19 @@ pub fn sum(array: &dyn Array) -> VortexResult<Scalar> {
         return Ok(Scalar::new(sum_dtype, sum));
     }
 
+    if array.is_empty() {
```

Comment:
Where was this broken? Or you're just lifting this logic into the entry point function?

Reply:
We got to this unreachable code https://github.com/spiraldb/vortex/blob/develop/vortex-array/src/arrays/bool/compute/sum.rs#L20

Reply:
Hmm, I need to figure out how we managed to do this